programs/zstdcli.c

```diff
@@ -672,7 +675,25 @@ int main(int argCount, const char* argv[])
         }
     }
 #endif
-
+    if (operation == zom_list) {
+        g_displayOut = stdout;
+        if(filenameIdx==0){
+            DISPLAY("No files given\n");
+            CLEAN_RETURN(0);
+        }
+        DISPLAY("===========================================\n");
+        DISPLAY("Printing information about compressed files\n");
+        DISPLAY("===========================================\n");
+        DISPLAY("Number of files listed: %d\n", filenameIdx);
+        {
+            int error = 0;
+            unsigned u;
+            for(u=0; u<filenameIdx;u++){
+                error = FIO_listFile(filenameTable[u],g_displayLevel);
```

Comment:
This will give you the last `error` of the last file.
If you want a non-zero value if "any" error happen, prefer `error +=` or `error |=`.